core/deployment/src/main/java/io/quarkus/deployment/builditem/ConnectionFactoryWrapperBuildItem.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkus.deployment.builditem;
+
+import java.util.function.Function;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+/**
+ * A build item that can be used to wrap ConnectionFactory
+ */
+public final class ConnectionFactoryWrapperBuildItem extends SimpleBuildItem {
+    private final Function<Object, Object> wrapper;
```

Comment:
How does an intended use look like? Is it really necessary that the wrapper holds a `Function<Object, Object>`, or can limit the bounds of this function somehow?

Reply:
Moreover, I think that the `Function` instance needs to be produced by a [recorder method](https://quarkus.io/guides/writing-extensions#bytecode-recording), or?

Reply:
Yeah, it is excatly we are doing in `quarkus-artemis` and `quarkus-pooled-jms`.